### ISSUE TITLE
[intersphinx] Cannot find reference targets for built-in exceptions when aliasing

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1233]
Info:
- Sphinx (sphinx-build) 1.2b1
- Python 2.6.6, Ubuntu
- Intersphinx enabled

This is similar to [Issue #1034](#1034), but it is a bit of a different use case.

My class has this:

```
#!python

class MyClass:
    failure_exception = AssertionError
```

My .rst file has this:

```
#!rst

.. automodule:: test.myclass
    :members:
```

And I get this exception:

```
Warning, treated as error:
/home/me/test/myclass.rst:35: WARNING: py:class reference target not found: AssertionError
```

When I don't use -W the docs show this (without the link to AssertionError):

```
class test.myclass.MyClass

    failure_exception
        alias of AssertionError
```

I have also tried adding this to the top, with no luck:

```
#!python

from exceptions import AssertionError
```

Is this a bug or am I doing it wrong?

Thanks,
Marc

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1233
- Originally reported by: [marc2982](https://bitbucket.org/marc2982)
- Originally created at: 2013-08-14T18:35:46.327

**Comment by shimizukawa** [comment]
_From [marc2982](https://bitbucket.org/marc2982) on 2013-08-14 22:27:46+00:00_

Quick update. I downloaded the objects.inv from Python2.6 and added the line:

```
#!python

AssertionError class library/exceptions.html
```

This allowed the link to be resolved. Of course, it links to http://docs.python.org/2.6/library/exceptions.html#AssertionError (which doesn't work), but it is close enough for my purposes as a workaround.

I hope this helps someone track this down.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-08-16 08:34:02+00:00_

This is correct entry for http://docs.python.org/2.6/objects.inv (it was already entried).

```
exceptions.AssertionError exception library/exceptions.html
```

Currently, displaying `alias of AssertionError` without the link to AssertionError is Sphinx specification I think.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-08-16 08:39:34+00:00_

small example

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-12 16:19:50+00:00_

Yes, since exceptions are also classes, the roles :exc: and :class: should be more or less interchangeable.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-12 16:20:43+00:00_

Closes #1233: Allow finding both Python classes and exceptions with the "class" and "exc" roles in intersphinx.

→ <<cset 12b3b49f2220207add2cce5713c072d5f66181e8>>
